src/main/java/com/rabbitmq/client/impl/recovery/TopologyRecoveryRetryLogic.java

```diff
@@ -62,7 +65,7 @@
         if (context.entity() instanceof RecordedQueue) {
             final RecordedQueue recordedQueue = context.queue();
             AutorecoveringConnection connection = context.connection();
-            connection.recoverQueue(recordedQueue.getName(), recordedQueue, false);
+          connection.recoverQueue(recordedQueue.getName(), recordedQueue);
```

Comment:
Before: if this failed to recover the queue, the error was delivered to the exception handler and swalled from the perspective of this retry logic. So the queue never got recovered.

Now: this recoverQueue method will now throw an exception if an error occurs and the retry logic will catch that and continue to retry as long as the max retry attempts haven't been reached.